app/Main.hs

```diff
@@ -235,7 +239,27 @@ getContent request = do
   -- Parse the response body as possibly malformed HTML and convert that to an XML tree.
   let doc = XML.documentRoot (DOM.parseLBS (HTTP.responseBody response))
       content = elementToContent doc
-  pure content
+  fetchImages (manager, request) content
+
+fetchImages :: (HTTP.Manager, HTTP.Request) -> Content -> IO Content
```

Comment:
To make this concurrent you could return a `Concurrently Content`: http://hackage.haskell.org/package/async-2.2.1/docs/Control-Concurrent-Async.html#t:Concurrently

and then `runConcurrently :: Concurrently Content -> IO Content`.

However, that would be unbounded concurrent connections. There is `pooledMapConcurrentlyN`,

http://hackage.haskell.org/package/unliftio-0.2.10/docs/UnliftIO-Async.html#v:pooledMapConcurrentlyN

but that only works on a traversable, and `Content` isn't that. 

There might be a good way to make a `Conc a` run in a bounded number of threads: http://hackage.haskell.org/package/unliftio-0.2.10/docs/UnliftIO-Async.html#t:Conc

Perhaps that can be modified so that `runConc` can take an additional arg like `runConcBounded :: Int -> Conc a -> IO a`.